### PR TITLE
Demo styles.

### PR DESCRIPTION
**Pull request by malcheg**
@Starzu check it please

### CODE REVIEW

File: core/src/main/scala/io/udash/generator/plugins/scalacss/ScalaCSSDemosPlugin.scala [hunk 1]
```diff
@@ -90,15 +215,15 @@ object ScalaCSSDemosPlugin extends GeneratorPlugin with SBTProjectFiles with Fro
...
+         |    h2(
          |      "You can find this demo source code in: ",
          |      i("${settings.rootPackage.mkPackage()}.${settings.viewsSubPackage.mkPackage()}.DemoStylesView")
          |    ),
          |    h3("Example"),
          |    p(LocalStyles.redItalic)("Red italic text."),
          |    p(LocalStyles.obliqueOnHover)("Hover me!"),
          |    h3("Read more"),
-         |    a(href := "http://japgolly.github.io/scalacss/book/")("Read more in ScalaCSS docs.")
+         |    a(DemoStyles.underlineLinkBlack)(href := "http://japgolly.github.io/scalacss/book/")("Read more in ScalaCSS docs.")
```
Comment (Starzu) [inline core/src/main/scala/io/udash/generator/plugins/scalacss/ScalaCSSDemosPlugin.scala]: We should add here also Udash Guide link: http://guide.udash.io/#/frontend/templates

File: core/src/main/scala/io/udash/generator/plugins/scalacss/ScalaCSSDemosPlugin.scala [hunk 2]
```diff
@@ -131,49 +256,1050 @@ object ScalaCSSDemosPlugin extends GeneratorPlugin with SBTProjectFiles with Fro
...
+         |jQ(DemoStyles.render[TypedTag[org.scalajs.dom.raw.HTMLStyleElement]].render).insertBefore(appRoot.get)
+         |jQ(FooterStyles.render[TypedTag[org.scalajs.dom.raw.HTMLStyleElement]].render).insertBefore(appRoot.get)
+         |jQ(HeaderStyles.render[TypedTag[org.scalajs.dom.raw.HTMLStyleElement]].render).insertBefore(appRoot.get)""".stripMargin)
+
+    writeFile(globalStylesScala)(
+      s"""package ${settings.rootPackage.mkPackage()}.${settings.stylesSubPackage.mkPackage()}
+          |
+          |import com.example.styles.constants.StyleConstants
+          |import com.example.styles.fonts.{FontStyle, FontWeight, UdashFonts}
+          |import com.example.styles.utils.{MediaQueries, StyleUtils}
```
Comment (Starzu) [inline core/src/main/scala/io/udash/generator/plugins/scalacss/ScalaCSSDemosPlugin.scala]: Replace `com.example` with `${settings.rootPackage.mkPackage()}`, please. Generated code does not compile, when you provide different root package.

**Comment by malcheg** [comment]
@Starzu take a look

**Comment by malcheg** [comment]
@Starzu updated

**Pull request merged**